ONPRC_EHR_ComplianceDB/resources/schemas/dbscripts/sqlserver/onprc_ehr_compliancedb-24.001-24.002.sql

```diff
@@ -0,0 +1,445 @@
+          )
+
+
+        If @@Error <> 0
+             GoTo Err_Proc
+   END
+
+
+ RETURN 0
+
+
+Err_Proc:
+                    -------Error Generated
+	RETURN 1
+
+
+END
+
+GO
+
+
+
+CREATE Procedure [onprc_ehr_compliancedb].[p_SciShieldHistoricalFinal]
+
+AS
+
+BEGIN
+             --- Process final historical set for Safety - Non-Human Primate (NHP) Annual Biosafety Training
+
+  If exists  (Select * from ehr_compliancedb.CompletionDates hh
+         where hh.requirementname =  'Safety - NHP Biosafety - Initial Training/Annual Refresher')
+     BEGIN
+
+            Update  hh
+                set hh.requirementname = 'Safety - Non-Human Primate (NHP) Annual Biosafety Training'
+          from ehr_compliancedb.CompletionDates hh where hh.requirementname =  'Safety - NHP Biosafety - Initial Training/Annual Refresher'
+
+
+                 If @@Error <> 0
+                        GoTo Err_Proc
+     END
+
+
+
+                  --- Process final historical set for Safety - Chemical Safety Training - Annual
+
+   If exists  (Select * from ehr_compliancedb.CompletionDates hh where hh.requirementname =  'Safety - Chemical Safety')
+    BEGIN
+
+            Update  hh
+                     set hh.requirementname = 'Safety - Chemical Safety Training - Annual'
+            from ehr_compliancedb.CompletionDates hh where hh.requirementname =  'Safety - Chemical Safety'
+
+                         If @@Error <> 0
+                                  GoTo Err_Proc
+   END
+
+
+            --- Process final historical set for Safety - Bioquell Z2 Vaporous Hydrogen Peroxide (VHP) Decon Unit
+
+ If exists  (Select * from ehr_compliancedb.CompletionDates hh where hh.requirementname =  'Safety - Bioquell Z2 Vaporous Hydrogen Peroxide (VHP) Decon Unit')
+  BEGIN
+          Update  hh
+              set hh.requirementname = 'Safety - Facility Decontamination Operator Training – Annual'
+
+         from ehr_compliancedb.CompletionDates hh where hh.requirementname =  'Safety - Bioquell Z2 Vaporous Hydrogen Peroxide (VHP) Decon Unit'
+
+                   If @@Error <> 0
+                         GoTo Err_Proc
+  END
+
+
+             --- Process final historical set for Safety - ClorDiSys Minidox Chlorine Dioxide Decon Unit
+
+ If exists  (Select * from ehr_compliancedb.CompletionDates hh where hh.requirementname =  'Safety - ClorDiSys Minidox Chlorine Dioxide Decon Unit')
+  BEGIN
+       update  hh
+            set hh.requirementname = 'Safety - Facility Decontamination Operator Training – Annual'
+       from ehr_compliancedb.CompletionDates hh where hh.requirementname =  'Safety - ClorDiSys Minidox Chlorine Dioxide Decon Unit'
+
+                  If @@Error <> 0
+                         GoTo Err_Proc
+  END
+
+
+RETURN 0
+
+
+Err_Proc:
+                    -------Error Generated
+	RETURN 1
+
+
+END
+
+GO
+
+
+
+
+CREATE TABLE [onprc_ehr_compliancedb].[SciShieldTemp] (
```

Comment:
@Ohsudev If you're planning on viewing these tables or using them at all in LK (not just your SP), I would recommend at least adding a PK and a container column. Most things in LK work best when there is a PK defined. And the container column is necessary to control permissions. You can see the other onprc_ehr_compliancedb script to see how to optimize the container column with FK and index. Let me know if you have any questions on that.